src/main/java/seedu/address/logic/commands/GetPrev3MonthFeeCommand.java

```diff
@@ -0,0 +1,83 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.fee.Month;
+import seedu.address.model.fee.Year;
+import seedu.address.model.session.Session;
+import seedu.address.model.student.Student;
+
+/**
+ * Gets current month and previous 2 months expected monthly fee based on the current list of sessions
+ */
+public class GetPrev3MonthFeeCommand extends Command {
+    public static final String COMMAND_WORD = "3Month_Fee";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Shows the total expected monthly fee"
+        + "for this month + previous 2 months";
+
+    /**
+     * Gets the local date time format of the month and year combined
+     * @return LocalDateTime of the month and year combined
+     */
+    private LocalDateTime getLocalDate(Month month, Year year) {
+        requireAllNonNull(month, year);
+        return LocalDateTime.of(year.getYear(), month.getMonth(), 1, 0, 0);
+    }
+
+    /**
+     * Gets the total fee between 2 time period {@code currMonthYear and nextMonthYear}
+     * @param studentList List of students
+     * @param startPeriod Start of time period
+     * @param endPeriod End of time period
+     * @return Total fee between the 2 time period
+     */
+    private double getFee(List<Student> studentList, LocalDateTime startPeriod, LocalDateTime endPeriod) {
+        double fee = 0;
+        for (Student student : studentList) {
+            for (Session session : student.getListOfSessions()) {
+                LocalDateTime dateTime = session.getSessionDate().getDateTime();
+                if (dateTime.compareTo(startPeriod) >= 0 && dateTime.compareTo(endPeriod) < 0) {
+                    // This session date is within the period
+                    fee += session.getFee().getFee();
+                }
+            }
+        }
+        return fee;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
```

Comment:
Im understanding that getPrev3 month is getMonth applied to every student, and for 3 months. I guess it'd be considered less "repeated code" if this method was built on getMonth. possibly using getMonth's methods from within getPrev3?

Reply:
Ok good suggestion, the only common place for both code base is on model. Hence, I have abstracted out the get fees code over to model. Do have a look, the output should still be the same. Thank you!

Reply:
super neat, hadn't thought of putting at model!

Reply:
super neat, hadn't thought of putting it at model!